integrations/Hotjar/browser.js

```diff
@@ -41,18 +45,31 @@ class Hotjar {
   }
 
   track(rudderElement) {
-    logger.debug("[Hotjar] track:: method not supported");
+    logger.debug("===In Hotjar track===");
+
+    const { event } = rudderElement.message;
+
+    if (!event) {
+      logger.error("Event name not present");
+      return;
+    }
+
+    // event name must not exceed 750 characters and can only contain alphanumeric, underscores, and dashes.
+    window.hj("event", event.substring(0, 750).replaceAll(" ", "_"));
```

Comment:
Consecutive whitespaces should be clubbed together. Otherwise, "hot   jar" becomes "hot___jar". Also, this needs to be done before the clipping to the threshold.

Reply:
resolved